src/zmq/zmqnotificationinterface.cpp

```diff
@@ -57,7 +58,10 @@ std::unique_ptr<CZMQNotificationInterface> CZMQNotificationInterface::Create(std
     {
         std::string arg("-zmq" + entry.first);
         const auto& factory = entry.second;
-        for (const std::string& address : gArgs.GetArgs(arg)) {
+        for (std::string& address : gArgs.GetArgs(arg)) {
+            // libzmq uses prefix "ipc://" for UNIX domain sockets
+            ReplaceAll(address, ADDR_PREFIX_UNIX, ADDR_PREFIX_IPC);
```

Comment:
Is `ReplaceAll` the right function to use here? 
Sure, it's unlikely for it to appear multiple times, but from a principle of least surprise angle, it'd make sense to only replace the prefix.